src/averageAges.js

```diff
@@ -55,7 +69,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const filteredPeople = people.filter(person => (
+    people.find(mother => mother.name === person.mother) && (onlyWithSon
```

Comment:
you use 2 different ways to write code to do the same things here and in `calculateWomenAverageAge` function (I mean handle optional flag in the filter). Both are good IMO (:+1: ) but choose one and use it everywhere to be consistent

Reply:
Also, `some` method fits here better than `find` - returns boolean